bindings/python/.evergreen/test.sh

```diff
@@ -11,55 +11,12 @@ MONGOCRYPT_DIR="$MONGOCRYPT_DIR"
 if [ "Windows_NT" = "$OS" ]; then # Magic variable in cygwin
     PYMONGOCRYPT_LIB=${MONGOCRYPT_DIR}/nocrypto/bin/mongocrypt.dll
     export PYMONGOCRYPT_LIB=$(cygpath -m $PYMONGOCRYPT_LIB)
-    # We need to create virtualenvs to install pymongocrypt's
-    # cffi, cryptography, and pymongo dependencies correctly from
-    # wheels. Otherwise, setup.py attempts and fails to install
-    # from the source distributions (.tar.gz).
-    VIRTUALENV="/cygdrive/c/python/Python27/python -m virtualenv"
-    $VIRTUALENV -p C:/python/Python27/python.exe venv27
-    PYTHON27="$(pwd)/venv27/Scripts/python"
-    # Upgrade pip to install the cryptography wheel and not the tar.
-    # <20.1 because 20.0.2 says a future release may drop support for 2.7.
-    $PYTHON27 -m pip install --upgrade 'pip<20.1'
-    # Upgrade setuptools because cryptography requires 18.5+.
-    # <45 because 45.0 dropped support for 2.7.
-    $PYTHON27 -m pip install --upgrade 'setuptools<45'
-    $PYTHON27 -m pip install . pymongo
-
-    $VIRTUALENV -p C:/python/Python34/python.exe venv34
-    PYTHON34="$(pwd)/venv34/Scripts/python"
-    # Upgrade pip to install the cryptography wheel and not the tar.
-    # <19.2 because 19.2 dropped support for 3.4.
-    $PYTHON34 -m pip install --upgrade 'pip<19.2'
-    $PYTHON34 -m pip install . pymongo
-
-    $VIRTUALENV -p C:/python/Python35/python.exe venv35
-    PYTHON35="$(pwd)/venv35/Scripts/python"
-    $PYTHON35 -m pip install --upgrade pip
-    $PYTHON35 -m pip install . pymongo
-
-    $VIRTUALENV -p C:/python/Python36/python.exe venv36
-    PYTHON36="$(pwd)/venv36/Scripts/python"
-    $PYTHON36 -m pip install --upgrade pip
-    $PYTHON36 -m pip install . pymongo
-
-    $VIRTUALENV -p C:/python/Python37/python.exe venv37
-    PYTHON37="$(pwd)/venv37/Scripts/python"
-    $PYTHON37 -m pip install --upgrade pip
-    $PYTHON37 -m pip install . pymongo
-
-    $VIRTUALENV -p C:/python/Python38/python.exe venv38
-    PYTHON38="$(pwd)/venv38/Scripts/python"
-    $PYTHON38 -m pip install --upgrade pip
-    $PYTHON38 -m pip install . pymongo
-
-    PYTHONS=("python" \
-             "$PYTHON27" \
-             "$PYTHON34" \
-             "$PYTHON35" \
-             "$PYTHON36" \
-             "$PYTHON37" \
-             "$PYTHON38")
+    PYTHONS=("C:/python/Python27/python.exe" \
```

Comment:
So now we just install everything in the base python environment? Shouldn't we use a virtualenv for this? (I almost feel like I saw a ticket about this recently).

Reply:
We don't install anything on the system. We just let `python setup.py test` fetch and locally install the dependencies automagically.

Reply:
Right. But I think this does mean that the deps are installed system-wide and _might_ still be present when/if this instance gets picked from the pool again to run the test-suite. Previously, we used a Python binary that resided inside a venv so cryptography and cffi were being installed in that virtualenv which would get wiped out with the rest of the test-run data (as each run gets a unique directory under /data/mci/... in which to run).

In any case, I doubt this is an actual problem but I'm just worried that we might run into some issues that are hard to debug/reproduce.

Reply:
Right. But I think this does mean that the deps are installed system-wide and _might_ still be present when/if this instance gets picked from the pool again to run the test-suite. Previously, we used a Python binary that resided inside a venv so cryptography and cffi were being installed in that virtualenv which would get wiped out with the rest of the test-run data (as each run gets a unique directory under /data/mci/... in which to run).

In any case, I doubt this is an actual problem but I'm just worried that we might run into some issues that are hard to debug/reproduce.

Reply:
I think if you'd like to ensure everything runs in a virtualenv, we can define the `$VIRTUALENV` variable for each platform and the `$BIN_DIR` variable (`Scripts` for WINDOWS and `bin` for other platforms) in the if...elif block and then modify the last codeblock as follows:

```
for PYTHON_BINARY in "${PYTHONS[@]}"; do
    # Clear cached eggs for different python versions.
    rm -rf .eggs
    rm -rf venv
    $VIRTUALENV -p $PYTHON_BINARY venv
    venv/$BIN_DIR/python -c 'import sys; print(sys.version)'
    venv/$BIN_DIR/python setup.py test
```

Reply:
I don't understand how polluting the system would be a concern with `python setup.py test` because it does not install anything on the system. If a library is needed it is fetched and cached in the local directory. If you think we should use a virtualenv for some other reason please open a new ticket.